index.js

```diff
@@ -0,0 +1,132 @@
+
+'use strict'
+const cote = require('cote')
+const u = require('@elife/utils')
+const axios = require('axios')
+const https = require('https')
+require('dotenv').config()
+
+
+var sModel = "text-davinci-003";
+var iMaxTokens = 2048;
+var sUserId = "1";
+var dTemperature = 0.5;  
+var OPENAI_API_KEY = process.env.CHATGPT_APIKEY;
+     
+const headers = {
+    'Content-Type': 'application/json',
+    "Accept": "application/json",
+    "Authorization": "Bearer " + OPENAI_API_KEY,
+
+}
+
+function main() {
+    startMicroservice()
+    registerWithCommMgr()
+}
+
+
+let msKey = 'everlife-chat-gpt'
+
+
+/*      outcome/
+ * Register ourselves as a message handler with the communication
+ * manager so we can handle requests for secret messages.
+ */
+function registerWithCommMgr() {
+    commMgrClient.send({
+        type: 'register-msg-handler',
+        mskey: msKey,
+        mstype: 'msg',
+        mshelp: [ { cmd: "/chatgpt", txt: `To receive replies from the OpenAI ChatGPT server without using any commands. Install this skill using '/install chatgpt' command`}],
+    }, (err) => {
+        if(err) u.showErr(err)
+    })
+}
+
+
+const commMgrClient = new cote.Requester({
+    name: 'ChatGPT Skill -> CommMgr',
+    key: 'everlife-communication-svc',
+})
+
+function sendReply(msg, req) {
+    req.type = 'reply'
+    req.msg = String(msg)
+    commMgrClient.send(req, (err) => {
+        if(err) u.showErr(err)
+    })
+}
+
+function startMicroservice() {
+
+    /*      understand/
+     * The microservice (partitioned by key to prevent
+     * conflicting with other services).
+     */
+    const svc = new cote.Responder({
+        name: 'Everlife ChatGPT Skill',
+        key: msKey,
+    })
+
+    /*      outcome/
+     * Respond to user messages asking us to code/decode things
+     */
+    svc.on('msg', (req, cb) => {
+        if(!OPENAI_API_KEY){
```

Comment:
I think we're misunderstanding each other here. I am suggesting if the OPENAI_API_KEY is not set that we simply don't activate the skill (or just log and error). Then we respond saying we aren't going to handle the message (`return cb()`) and the remaining brains will take over as before.